File: 0136-single-number/0136-single-number.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        diction = dict()
        
        for n in nums:
            if n not in diction:
                diction[n] = 1
            else:
                diction[n] += 1
                
        for d in diction:
            if diction[d] == 1:
                return d

File: 0136-single-number/0136-single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        Arrays.sort(nums);
        int i = 0;
        while(i < nums.length-1){
            if(nums[i] != nums[i + 1]){
                return nums[i];
            }
            else{
                i += 2;
            }
        }
        
        return nums[nums.length-1];
    }
}